### ISSUE TITLE
Add aria-label for Close (X) icon in Dialog

### ISSUE THREAD

**Issue by vaibhavpaliwal** [cerner/terra-core#1351]
# Issue Description
Html code sniffer with WCAG2 shows below error about accessibility
![image](https://user-images.githubusercontent.com/8019344/37494661-dad19a5c-2877-11e8-80ed-839c67dc6280.png)

## Issue Type
- [x] Enhancement

## Expected Behavior
Aria-label should be present so there is no accessibility error.

## Steps to Reproduce
- Go to terra site dialog example page and run the html code sniffer [extension](http://squizlabs.github.io/HTML_CodeSniffer/)

## Environment

* Component Version: Dialog Version: 1.2.0
* Browser Name and Version: Chrome Version 65.0.3325.146
* Operating System and version (desktop or mobile): desktop

**Comment by zplata** [comment]
Related to the Close (X) button in Dialog, I'm seeing some errors when the Dialog renders due to the prop `text` being of value `undefined` on that Close `Button` component. I also noticed that issue in the source code of the Dialog component [here](https://github.com/cerner/terra-core/blob/master/packages/terra-dialog/src/Dialog.jsx#L43). 

<img width="1422" alt="screen shot 2018-03-28 at 4 23 11 pm" src="https://user-images.githubusercontent.com/5685058/38057412-fb23237a-32a4-11e8-85f8-67acb9f75290.png">
